DataDog/datadog-agent
# upgrade(metadata): Support Fargate agents in metadata payloads

### What does this PR do?
This is a first step to support Fargate agents metadata. 
Adds the UUID field present in the v5 metadata payload to:
- Inventories payloads
- `agent_check` payload
- Remote Config payloads

And adds a ECS Fargate section field in the `inventory_agent` payload, containing `ecs_fargate_task_arn` & `ecs_fargate_cluster_name`

### Motivation
[RC-1378]

### Additional Notes
See [RFC](https://docs.google.com/document/d/1ksCGrsFtSrrCMTOjIFmNrIbxRzlPU9P7ETQWJ2842fU/edit?pli=1)

### Possible Drawbacks / Trade-offs
The ECS payload part has not been tested as I couldn't build a new image

### Describe how to test/QA your changes

#### UUID in payloads
- Build & run your agent
- Run the `agent diagnose show-metadata <payload>` command for every payload
- Check that UUID is here, properly filled, and stable

#### ECS Fargate `inventory_agent` Payload
- Update the image in the [remote-config-agent-test](https://us-east-1.console.aws.amazon.com/ecs/v2/task-definitions/remote_config_agent_test?region=us-east-1) Fargate cluster in the Sandbox AWS account
- Set the latest RC as the image
- Connect to your container using `aws ecs execute-command  --region us-east-1 --cluster remote-config-agent-test --task <task_id> --container datadog-agent --command "/bin/bash" --interactive`
- Run the `agent diagnose show-metadata inventory-agent` command, check if the `ecs_fargate_task_arn` & `ecs_fargate_cluster_name` are there and properly filled

### Reviewer's Checklist
- [x] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.


[RC-1378]: https://datadoghq.atlassian.net/browse/RC-1378?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
